lib/tasks/precompute_gene_ontology.rake

```diff
@@ -0,0 +1,209 @@
+MASTER_OWL_URI = "https://raw.githubusercontent.com/arpcard/aro/master/aro.owl".freeze
+URL_PUBMED = "https://www.ncbi.nlm.nih.gov/pubmed/".freeze
+S3_JSON_BUCKET = "idseq-database".freeze
+S3_ARGANNOT_FASTA = "s3://idseq-database/amr/ARGannot_r2.fasta".freeze
+
+DRUG_CLASSES = {
+  "agly" => ["Aminoglycosides", "0000016"],
+  "bla" => ["Beta-lactamases", "3000001"],
+  "colistin" => ["Colistins", "0000067"],
+  # "fcd" => [], # Unclear which one this refers to
+  "fos" => ["Fosfomycin", "0000025"],
+  "fcyn" => ["Fosfomycin", "0000025"],
+  "flq" => ["Fluroquinolones", "0000001"],
+  "gly" => ["Glycopeptides", "3000081"],
+  "mls" => ["Macrolide-lincosamide-streptogramin", "0000000"],
+  "ntmdz" => ["Nitroimidazoles", "3004115"],
+  "oxzln" => ["Oxazolidinones", "3000079"],
+  "phe" => ["Phenicols", "3000387"],
+  "rif" => ["Rifampicin", "3000169"],
+  "sul" => ["Sulfonamides", "3000282"],
+  "tet" => ["Tetracyclines", "3000050"],
+  "tmt" => ["Trimethoprim", "3000188"],
+}.freeze # label, ARO Accession number
+
+require "aws-sdk-s3"
+require "nokogiri"
+
+desc 'Updates and precomputes AMR Ontology information'
+task precompute_gene_ontology: :environment do
+  s3 = Aws::S3::Client.new(region: DEFAULT_S3_REGION)
+
+  owl_uri = URI(MASTER_OWL_URI)
+  response = Net::HTTP.get_response(owl_uri)
+  unless response.is_a?(Net::HTTPSuccess)
+    Rails.logger.error("Unable to fetch OWL file")
+    return
+  end
+  owl_doc = Nokogiri::XML(response.body.force_encoding("utf-8"))
+
+  arg_annot_fasta = nil
+  split_uri = S3_ARGANNOT_FASTA.split("/", 4)
+  bucket = split_uri[2]
+  key = split_uri[3]
+  begin
+    resp = s3.get_object(bucket: bucket, key: key)
+    arg_annot_fasta = resp.body.read
+  rescue Aws::S3::Errors => err
+    Rails.logger.error("Unable to fetch ARG ANNOT fasta file from S3")
+    Rails.logger.error(err.message)
+    return
+  end
+
+  split_lines = arg_annot_fasta.split("\n")
+  arg_entries = []
+  split_lines.each do |line|
+    if line[0] == ">"
+      split_entry = line.split(";")
+      gene_name = split_entry[2]
+      drug_class = if DRUG_CLASSES.key?(drug_class)
```

Comment:
Is there a line missing here? `drug_class` doesn't seem to be defined before this line.

Reply:
That's a Ruby conditional assignment! This is where `drug_class` is defined. If you don't like it, take it up with Rubocop 🚨haha